bundle/edu.gemini.spModel.core/src/main/java/edu/gemini/spModel/core/Semester.java

```diff
@@ -156,6 +156,13 @@ public Date getStartDate(Site site) {
         return cal.getTime();
     }
 
+    public Date getMidpointDate(Site site) {
+        final long s = getStartDate(site).getTime();
+        final long e = getEndDate(site).getTime();
+        final long m = s + (e - s)/2;
+        return new Date(m);
+    }
```

Comment:
The P1 `Target` -> `SPTarget` converter needs a time for nonsidereal targets.  The skeleton creator uses the semester midpoint, so that's what I did too.